sdk/storage/Azure.Storage.Files.DataLake/src/PathClient.cs

```diff
@@ -0,0 +1,1462 @@
+
+                    uriBuilder.DirectoryOrFilePath = destinationPath;
+                    PathClient destPathClient = new PathClient(uriBuilder.ToUri(), Pipeline);
+
+                    Response<PathCreateResult> response = await DataLakeRestClient.Path.CreateAsync(
+                        clientDiagnostics: _clientDiagnostics,
+                        pipeline: Pipeline,
+                        resourceUri: destPathClient.DfsUri,
+                        mode: PathRenameMode.Legacy,
+                        renameSource: renameSource,
+                        leaseId: destConditions?.LeaseId,
+                        sourceLeaseId: sourceConditions?.LeaseId,
+                        ifMatch: destConditions?.IfMatch,
+                        ifNoneMatch: destConditions?.IfNoneMatch,
+                        ifModifiedSince: destConditions?.IfModifiedSince,
+                        ifUnmodifiedSince: destConditions?.IfUnmodifiedSince,
+                        sourceIfMatch: sourceConditions?.IfMatch,
+                        sourceIfNoneMatch: sourceConditions?.IfNoneMatch,
+                        sourceIfModifiedSince: sourceConditions?.IfModifiedSince,
+                        sourceIfUnmodifiedSince: sourceConditions?.IfUnmodifiedSince,
+                        async: async,
+                        cancellationToken: cancellationToken)
+                        .ConfigureAwait(false);
+
+                    return Response.FromValue(
+                        destPathClient,
+                        response.GetRawResponse());
+                }
+                catch (Exception ex)
+                {
+                    Pipeline.LogException(ex);
+                    throw;
+                }
+                finally
+                {
+                    Pipeline.LogMethodExit(nameof(PathClient));
+                }
+            }
+        }
+        #endregion Rename
+
+        #region Get Access Control
+        /// <summary>
+        /// The <see cref="GetAccessControlInternal"/> operation returns the
+        /// access control data for a path.
+        ///
+        /// For more information, see <see href="https://docs.microsoft.com/en-us/rest/api/storageservices/datalakestoragegen2/path/getproperties" />.
+        /// </summary>
+        /// <param name="upn">
+        /// Optional.Valid only when Hierarchical Namespace is enabled for the account.If "true",
+        /// the user identity values returned in the x-ms-owner, x-ms-group, and x-ms-acl response
+        /// headers will be transformed from Azure Active Directory Object IDs to User Principal Names.
+        /// If "false", the values will be returned as Azure Active Directory Object IDs.The default
+        /// value is false. Note that group and application Object IDs are not translated because they
+        /// do not have unique friendly names.
+        /// </param>
```

Comment:
Is there a longer name we can use for these?

Reply:
I don't actually know what upn stands for, but we can look into it.

Reply:
I _think_ it's `UserPrincipalName`

Reply:
https://github.com/Azure/azure-sdk-for-net/issues/8391